packages/story-editor/src/components/panels/design/pageAttachment/pageAttachment.js

```diff
@@ -301,25 +271,36 @@ function PageAttachmentPanel() {
       title={__('Call to Action', 'web-stories')}
       collapsedByDefault={false}
     >
-      <LinkInput
-        onChange={(value) => handleChangeUrl(value?.trim())}
-        onBlur={() => {
-          debouncedUpdate.cancel();
-          updatePageAttachment({
-            url: _url?.trim().length ? withProtocol(_url.trim()) : '',
-          });
-        }}
-        onFocus={onFocus}
-        value={_url}
-        clear
-        aria-label={__(
-          'Type an address to add a page attachment link',
-          'web-stories'
-        )}
-        hasError={hasError}
-        hint={hint}
-      />
-      {hasValidUrl && (
+      {hasProducts ? (
```

Comment:
As per the original ticket description, this whole component/file is becoming rather long.
And now with these ternaries everywhere it's becoming a bit more difficult to see at a glance what UI options are available for each type of page attachment (regular links vs shopping attachment).

This is a good opportunity to consider extracting the components into separate files (and also a good opportunity to add tests while at it). WDYT?

Can also create a new ticket for that of course, but now seems a good opportunity to do it in one go.

Reply:
As you can see, I did refactor this component to make it much smaller. Lots of repeated code has been removed. How would think to refactor, break it into multiple compontents?

Reply:
>  break it into multiple compontents?

Yes, that's what I meant by "extracting the components into separate files"